Contributor's Quick Guide (#12346)
mistercrunch/Airflow: docs/contributrs-quick-start.rst

 .. Licensed to the Apache Software Foundation (ASF) under one
    or more contributor license agreements.  See the NOTICE file
    distributed with this work for additional information
    regarding copyright ownership.  The ASF licenses this file
    to you under the Apache License, Version 2.0 (the
    "License"); you may not use this file except in compliance
    with the License.  You may obtain a copy of the License at

 ..   http://www.apache.org/licenses/LICENSE-2.0

 .. Unless required by applicable law or agreed to in writing,
    software distributed under the License is distributed on an
    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, either express or implied.  See the License for the
    specific language governing permissions and limitations
    under the License.

*************************
Contributor's Quick Guide
*************************

.. contents:: :local:

Prerequisites
#############

> 1. Docker Community Edition
> 2. Docker Compose
> 3. pyenv (you can also use pyenv-virtualenv or virtualenvwrapper)

Reviewer: ```suggestion
* Docker Community Edition (Linux)
* Docker Desktop (Windows/Mac OS)
* Docker Compose
* pyenv (you can also use pyenv-virtualenv or virtualenvwrapper)
```

Reviewer: Can you also add docs for Windows/Mac OS (not Apple Silicon based 🤣)?

Author: I did try to setup on windows and mac in VM but due to docker virtualisation error i could not try. And on windows also i was not able to setup pyenv virtualenv.

![DockerError](https://user-images.githubusercontent.com/11897651/99148486-ef3ecc80-26ad-11eb-804f-024f91a0716f.png)
![WinError](https://user-images.githubusercontent.com/11897651/99148492-f5cd4400-26ad-11eb-8341-4eac7df91a96.png)

Installing Prerequisites on Ubuntu
###################################

> Docker Community Edition

Reviewer: ```suggestion
Docker Community Edition (Linux)
```

-----------------------------------

> 1. Installing required packages for Docker and setting up docker repo

Reviewer: ```
curl -fsSL https://get.docker.com | bash -
```
Is just one command not enough?

.. code-block:: bash

  $ sudo apt-get update

  $ sudo apt-get install \
      apt-transport-https \
      ca-certificates \
      curl \
      gnupg-agent \
      software-properties-common

  $ curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -

  $ sudo add-apt-repository \
     "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
     $(lsb_release -cs) \
     stable"

2. Install Docker

.. code-block:: bash

  $ sudo apt-get update
  $ sudo apt-get install docker-ce docker-ce-cli containerd.io

3. Creating group for docker and adding current user to it.

.. code-block:: bash

  $ sudo groupadd docker
  $ sudo usermod -aG docker $USER

Note : After adding user to docker group Logout and Login again for group membership re-evaluation.

4. Test Docker installation

.. code-block:: bash

  $ docker run hello-world

> 

Reviewer: ```suggestion
For more information about instalation on Linux, see: ...
```

Docker Compose
--------------------------

1. Check the latest version of Docker Compose version |here| and Install

.. |here| raw:: html

   <a href="https://docs.docker.com/compose/install/#install-compose-
   on-linux-systems">here</a>

.. code-block:: bash

>   $ sudo curl -L "https://github.com/docker/compose/releases/download/1.27.4/docker-compose-$(uname -s)-$(uname -m)" \

Reviewer: ```suggestion
  $ COMPOSE_VERSION="$(curl -s https://api.github.com/repos/docker/compose/releases/latest | grep '"tag_name":' | cut -d '"' -f 4)"
  $ COMPOSE_URL="https://github.com/docker/compose/releases/download/${COMPOSE_VERSION}/docker-compose-$(uname -s)-$(uname -m)"
  $ sudo curl -L "${COMPOSE_URL}" \
```
Creating new variables makes it easier to edit the script, e.g. when you want to download different version.

Author: Nice, i will add this in next commit.

Reviewer 2: I wonder if it would be cleaner to use `jq`, but it may require the user to preinstall it (which I think it's fine because a lot of people use it?)

```
curl -s https://api.github.com/repos/docker/compose/releases/latest | jq '.tag_name' -r
```

Reviewer: @xinbinhuang Does jq give us any more benefits apart from a minimally more readable code?  I would like to use jq if we were to perform a bit more complex operations. For now the standard linux tools are sufficient.

Reviewer 2: I think my comment is only a nitpicking: The only benefit here is more readable code in my opinion. Other than that, they perform the same function. I agree that it's good for now
